File: src/main/java/com/dh/ondot/schedule/api/response/PlaceHistoryResponse.java
package com.dh.ondot.schedule.api.response;

import com.dh.ondot.core.util.TimeUtils;
import com.dh.ondot.schedule.domain.PlaceHistory;

import java.time.LocalDateTime;
import java.util.List;

public record PlaceHistoryResponse(
        String title,
        String roadAddress,
        Double longitude,
        Double latitude,
        LocalDateTime searchedAt
) {
    public static PlaceHistoryResponse from(PlaceHistory history) {
        return new PlaceHistoryResponse(
                history.getTitle(),
                history.getRoadAddress(),
                history.getLongitude(),
                history.getLatitude(),
                TimeUtils.toSeoulDateTime(history.getSearchedAt())
        );
    }
    public static List<PlaceHistoryResponse> fromList(List<PlaceHistory> list) {
        return list.stream().map(PlaceHistoryResponse::from).toList();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/presentation/response/PlaceHistoryResponse.kt
package com.dh.ondot.schedule.presentation.response

import com.dh.ondot.core.util.TimeUtils
import com.dh.ondot.schedule.domain.PlaceHistory
import java.time.LocalDateTime

data class PlaceHistoryResponse(
    val title: String?,
    val roadAddress: String?,
    val longitude: Double?,
    val latitude: Double?,
    val searchedAt: LocalDateTime?,
) {
    companion object {
        @JvmStatic
        fun from(history: PlaceHistory): PlaceHistoryResponse {
            return PlaceHistoryResponse(
                title = history.title,
                roadAddress = history.roadAddress,
                longitude = history.longitude,
                latitude = history.latitude,
                searchedAt = TimeUtils.toSeoulDateTime(history.searchedAt),
            )
        }

        @JvmStatic
        fun fromList(list: List<PlaceHistory>): List<PlaceHistoryResponse> {
            return list.map { from(it) }
        }
    }
}
